assets/sass/_theme/sections/locations.sass

```diff
@@ -1,51 +1,86 @@
 .locations
+    display: flex
+    flex-direction: column
+    gap: $spacing-4 0
+    .location
+        cursor: pointer
+        display: flex
+        flex-direction: column-reverse
+        gap: $spacing-3 var(--grid-gutter)
+        justify-content: flex-end
+        position: relative
+        a
+            text-decoration: none
+        .more
+            @include meta
+            margin-top: $spacing-1
+            @include stretched-link(before)
+            @include icon(arrow-right-line, after)
+                margin-left: $spacing-2
+            @include hover-translate-icon(after)
+
+        .media
+            aspect-ratio: 3/2
+            background: var(--color-background-alt)
+            height: fit-content
+            width: columns(6)
+            picture
+                aspect-ratio: 3/2
+                display: block
+            img
+                display: block
+                height: 100%
+                object-fit: cover
+        &-description
+            margin-top: $spacing-1
     &--list
-        .location + .location
-            margin-top: $spacing-4
         @include media-breakpoint-up(desktop)
             .location
-                flex-direction: row
-                .media
+                flex-direction: row-reverse
+                .location-content
+                    display: flex
+                    flex-direction: column
+                    justify-content: space-between
+                .location-description
+                    display: flex
+                    flex-direction: column
+                    gap: $spacing-1
+                    height: 100%
+                    justify-content: space-between
                     width: columns(4)
-                &-content
-                    width: columns(6)
+                @include in-page-with-sidebar
+                    .location-description
+                        width: columns(6)
+                    .media
+                        width: columns(2)
+                @include in-page-without-sidebar
```

Comment:
Tu peux tout à fait simplifier ce qu'il y a dans ton `in-page-without-sidebar` en appliquant un `grid(2)` sur ton `.location-content`, associé à un `flex: 1`. Le `flex: 1` fait prendre au content toute la largeur restante en dehors de celle de `.media`, sans que tu aies besoin de préciser la largeur. Le grid fait le reste.

Pour être + compréhensible : 
<img width="952" alt="Capture d’écran 2024-09-02 à 11 04 39" src="https://github.com/user-attachments/assets/eca7ff0c-e48a-4c7a-b3c1-9aafd76f9a5e">